packages/web-components/src/components/ic-dialog/ic-dialog.tsx

```diff
@@ -464,6 +465,33 @@ export class Dialog {
     }
   }
 
+  private fixSlottedTooltip = (element: Element) => {
```

Comment:
just a slight optimisation to avoid unnecessary code execution \ variable assignment
```suggestion
 private fixSlottedTooltip = (element: Element) => {
    let tooltip: HTMLIcTooltipElement | null = null;
    if (element.tagName === "IC-TOOLTIP") {
      tooltip = element as HTMLIcTooltipElement;
    } else {
      const tooltipChildElement =
      element.shadowRoot?.querySelector<HTMLIcTooltipElement>("ic-tooltip");
    
      if (tooltipChildElement) {
        tooltip = tooltipChildElement;
      } else if (element.children?.length > 0) {
        Array.from(element.children).forEach((child) =>
          this.fixSlottedTooltip(child)
        );
      } else {
        return;
      }
  }
    tooltip?.setExternalPopperProps({ strategy: "fixed" });
  };
```